spec/requests/admin_requests_spec.rb

```diff
@@ -12,19 +12,18 @@
     context "with rendered views" do
       let!(:users_list) { create_list(:user, 25) }
       let!(:user) { create(:user, name: "Name Not Provided") }
-      edit_user_path_regex = Regexp.new('admin/users/[0-9]+/edit\\?organization_id=admin')
 
       it "shows a logout button" do
         get admin_dashboard_path
         expect(response.body).to match(/log out/im)
       end
 
-      it "shows the recently added users email " do
+      it "shows the recently added users email" do
         get admin_dashboard_path
-
-        expect(response.body).to match(/20 New users/im)
-        expect(response.body).to match(/#{user.email}/im)
-        expect(response.body).to match(edit_user_path_regex)
+        expect(response).to have_http_status(:success)
+        expect(response.body).to match(/20 New Users/im)
+        expected_path = edit_admin_user_path(user, organization_id: "admin")
+        expect(response.body).to include(CGI.escapeHTML(expected_path))
       end
```

Comment:
The changes you made to this test make it not check the email.   I suggest going back to the original test -- except that the user should have a nil name, right?

Reply:
Hello @cielf. I understand that testing the email is our priority when implementing this test. But at the same time, I checked that the Dashboard view doesn't show email on screen, there is an email being used just to bring other information.

I think there are 2 ways:
1) Implementing a change to the dashboard to show user emails. But that probably would require an extra Issue, I guess. I went all the way on this and have tested adding this line to the dashboard and it test passed using the old test (I also added user null as you suggested): 

<div class="users-list-email"><%= user.email %></div>.

/app/view/admin/dashboard.html.erb after line 97.

Then bundle exec rspec ./spec/requests/admin_requests_spec.rb works without changing the original test.

2) Not considering testing the email, since this test was passing (as "False-Positive" test before) because the view doesn't show the email. In that case, I would need to remove the line that tests the email from the original test, since it will not find any email in the view as commented before.

Hope I made myself understandable. I look up to your guidance for the resolution of this case.

Reply:
The dashboard currently shows the user email if and only if the user name is "Name Not Provided".  Since we are reworking to get rid of "Name Not Provided" that check should be for "Name Not Provided" or nil.   But we still need the email to be shown in these cases so that we can differentiate between any users who have not provided a name.

Reply:
Thanks @cielf I didn't think of this case you described. So I implemented a new set of changes to the proposed code included in this PR.